### PR TITLE
[Hub Generated] Review request for Microsoft.DataLakeAnalytics to add version preview/2015-10-01-preview

### PR DESCRIPTION
**Pull request by sravan251**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/sravan251/azure-rest-api-specs/adlaswaggerspecsfix...master/).

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “**WaitForARMFeedback**” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [x] Updating API in stable version with Breaking Change Validation errors
- [x] Updating API(s) in preview over 1 year

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by sravan251** [comment]
@njuCZ can you please review the pr. There are some SDK failures and i am not sure how to fix them

File: specification/datalake-analytics/resource-manager/Microsoft.DataLakeAnalytics/stable/2016-11-01/account.json [hunk 1]
```diff
@@ -242,6 +242,15 @@
             "schema": {
               "$ref": "#/definitions/DataLakeAnalyticsAccount"
             }
+          },
+          "409": {
```
Comment (majastrz) [inline specification/datalake-analytics/resource-manager/Microsoft.DataLakeAnalytics/stable/2016-11-01/account.json:246]: >409 [](start = 11, length = 3)

This isn't the complete list of the responses you can return. What about 500, 503, 504, 499? Also, the error response is not RPC-compliant. (Take a look at [https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/common-api-details.md#error-response-content](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/common-api-details.md#error-response-content) for details.)

Instead of listing every possible failure, you should simply declare a "default" response as a catch-all for all errors with a response that matches the link I provided above.

Comment (majastrz) [reply specification/datalake-analytics/resource-manager/Microsoft.DataLakeAnalytics/stable/2016-11-01/account.json:246]: The comment applies to all similar changes below.

---
In reply to: [513045359](https://github.com/Azure/azure-rest-api-specs/pull/11289#discussion_r513045359) [](ancestors = 513045359)

**Comment by majastrz** [comment]
>     "/subscriptions/{subscriptionId}/providers/Microsoft.DataLakeAnalytics/accounts": {

Why is the diff in this file so messy compared to the other API version?  Can you summarize the changes to this API version?

---
Refers to: specification/datalake-analytics/resource-manager/Microsoft.DataLakeAnalytics/preview/2015-10-01-preview/account.json:40 in e15ce8e. [](commit_id = e15ce8e19c5e85dbbcdc3f2e21b5cc706e118c35, deletion_comment = False)

**Comment by sravan251** [comment]
> > ```
> > "/subscriptions/{subscriptionId}/providers/Microsoft.DataLakeAnalytics/accounts": {
> > ```
> 
> Why is the diff in this file so messy compared to the other API version? Can you summarize the changes to this API version?
> 
> Refers to: specification/datalake-analytics/resource-manager/Microsoft.DataLakeAnalytics/preview/2015-10-01-preview/account.json:40 in [e15ce8e](https://github.com/Azure/azure-rest-api-specs/commit/e15ce8e19c5e85dbbcdc3f2e21b5cc706e118c35). [](commit_id = [e15ce8e](https://github.com/Azure/azure-rest-api-specs/commit/e15ce8e19c5e85dbbcdc3f2e21b5cc706e118c35), deletion_comment = False)



> > ```
> > "/subscriptions/{subscriptionId}/providers/Microsoft.DataLakeAnalytics/accounts": {
> > ```
> 
> Why is the diff in this file so messy compared to the other API version? Can you summarize the changes to this API version?
> 
> Refers to: specification/datalake-analytics/resource-manager/Microsoft.DataLakeAnalytics/preview/2015-10-01-preview/account.json:40 in [e15ce8e](https://github.com/Azure/azure-rest-api-specs/commit/e15ce8e19c5e85dbbcdc3f2e21b5cc706e118c35). [](commit_id = [e15ce8e](https://github.com/Azure/azure-rest-api-specs/commit/e15ce8e19c5e85dbbcdc3f2e21b5cc706e118c35), deletion_comment = False)

The preview version is introduced in 2015 never updated the swagger specs after that. right now its similar to 2016-11-01 stable version in all operations and response.

File: specification/datalake-analytics/resource-manager/Microsoft.DataLakeAnalytics/stable/2016-11-01/account.json (hunk 1 above)
Comment (sravan251) [reply specification/datalake-analytics/resource-manager/Microsoft.DataLakeAnalytics/stable/2016-11-01/account.json:246]: Added the default response for all paths and responses

**Comment by majastrz** [comment]
> The preview version is introduced in 2015 never updated the swagger specs after that. right now its similar to 2016-11-01 stable version in all operations and response.

So this update to `2015-10-01-preview` swagger makes it match the behavior of your backend when it's handling requests with API version `2015-10-01-preview`. Is that correct?

File: specification/datalake-analytics/resource-manager/Microsoft.DataLakeAnalytics/stable/2016-11-01/account.json (hunk 1 above)
Comment (majastrz) [reply specification/datalake-analytics/resource-manager/Microsoft.DataLakeAnalytics/stable/2016-11-01/account.json:246]: Did you push the commit? The last commit I'm seeing is from ~8 hours ago.

File: specification/datalake-analytics/resource-manager/Microsoft.DataLakeAnalytics/stable/2016-11-01/account.json [hunk 2]
```diff
@@ -1564,6 +1762,21 @@
...
+    "ErrorContract": {
+      "title": "Error details.",
+      "description": "Contains details when the response code indicates an error.",
+      "type": "object",
+      "properties": {
+        "error": {
+          "description": "The error details.",
+          "allOf": [
+            {
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
```
Comment (majastrz) [inline specification/datalake-analytics/resource-manager/Microsoft.DataLakeAnalytics/stable/2016-11-01/account.json:1774]: >ErrorResponse [](start = 98, length = 13)

v2 error response already defines the `error` property at the top level ([https://github.com/Azure/azure-rest-api-specs/blob/master/specification/common-types/resource-management/v2/types.json](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/common-types/resource-management/v2/types.json)), which would make this look like the following in the json:
```json
"error": {
  "error": {
    ...
  }
}
```

Instead of referencing `#/definitions/ErrorContract` you should reference `../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse` in all the default responses.

File: specification/datalake-analytics/resource-manager/Microsoft.DataLakeAnalytics/stable/2016-11-01/account.json (hunk 1 above)
Comment (sravan251) [reply specification/datalake-analytics/resource-manager/Microsoft.DataLakeAnalytics/stable/2016-11-01/account.json:255]: yes pushed

**Comment by majastrz** [comment]
>           "409": {

Same comments here as in the other file.

---
Refers to: specification/datalake-analytics/resource-manager/Microsoft.DataLakeAnalytics/preview/2015-10-01-preview/account.json:246 in 084f244. [](commit_id = 084f244d8049524196e78473453d6dac211e524d, deletion_comment = False)

**Comment by sravan251** [comment]
> > The preview version is introduced in 2015 never updated the swagger specs after that. right now its similar to 2016-11-01 stable version in all operations and response.
> 
> So this update to `2015-10-01-preview` swagger makes it match the behavior of your backend when it's handling requests with API version `2015-10-01-preview`. Is that correct?

yes, thats correct

**Comment by majastrz** [comment]
>     "ErrorContract": {

Same issue here as well as in the other file.

---
Refers to: specification/datalake-analytics/resource-manager/Microsoft.DataLakeAnalytics/preview/2015-10-01-preview/account.json:1643 in 084f244. [](commit_id = 084f244d8049524196e78473453d6dac211e524d, deletion_comment = False)

File: specification/datalake-analytics/resource-manager/Microsoft.DataLakeAnalytics/stable/2016-11-01/account.json (hunk 2 above)
Comment (sravan251) [reply specification/datalake-analytics/resource-manager/Microsoft.DataLakeAnalytics/stable/2016-11-01/account.json]: fixed.

**Comment by majastrz** [comment]
>           "400": {

What about this one?

---
Refers to: specification/datalake-analytics/resource-manager/Microsoft.DataLakeAnalytics/preview/2015-10-01-preview/account.json:798 in 8b2794e. [](commit_id = 8b2794e5ad2a3da52e2f71563432797d86acac6c, deletion_comment = False)

**Comment by majastrz** [comment]
>           "410": {

And this one

---
Refers to: specification/datalake-analytics/resource-manager/Microsoft.DataLakeAnalytics/preview/2015-10-01-preview/account.json:994 in 8b2794e. [](commit_id = 8b2794e5ad2a3da52e2f71563432797d86acac6c, deletion_comment = False)

**Review by majastrz — APPROVED** [review]
Signed off on the swagger bug fixes from the ARM side. You should still decide if you really want to enumerate some of the error codes in the responses. There is impact to the generated SDKs and most RPs don't list any codes and only have the default. The reviewer from the SDK team can also advise you here, but you should make sure that you're getting the behavior that you need in your SDKs.

**Comment by majastrz** [comment]
Looks good. Still signed off 🙂

**Comment by sravan251** [comment]
@njuCZ please review at earliest. We have many voilations that are going out of SLA

**Comment by sravan251** [comment]
@njuCZ please provide feedback. this change reruired to go

**Comment by sravan251** [comment]
@njuCZ please review

**Comment by njuCZ** [comment]
@sravan251 could you please fix the `PrettierCheck` CI first?

**Comment by sravan251** [comment]
@njuCZ i have fixed the Prettier validation. Please review

**Comment by sravan251** [comment]
> @sravan251 could you please fix the `PrettierCheck` CI first?

fixed.please review

**Comment by njuCZ** [comment]
@sravan251 sorry for late reply. Could you please fix the lintDiff CI error? and it seems there are some rest api deleted? Could I ask why made these changes?

**Comment by sravan251** [comment]
@njuCZ Can you please review the Changes. I am working with ARM reivew Board for review. Please post if you have any specific questions.

**Comment by akning-ms** [comment]
Hi @sravan251 , you may need to fix below semantic error first:

EQUIVALENT_PATH | Equivalent path already exists: /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DataLakeAnalytics/accounts/{name}JsonUrl: Microsoft.DataLakeAnalytics/preview/2015-10-01-preview/account.json#L312
-- | --

File: specification/datalake-analytics/resource-manager/Microsoft.DataLakeAnalytics/preview/2015-10-01-preview/account.json [hunk 3]
```diff
@@ -41,509 +37,659 @@
...
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-odata": "#/definitions/DataLakeAnalyticsAccountBasic"
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DataLakeAnalytics/accounts/{accountName}": {
+      "get": {
```
Comment (akning-ms) [inline specification/datalake-analytics/resource-manager/Microsoft.DataLakeAnalytics/preview/2015-10-01-preview/account.json:216]: it still not fix duplicate path name issue.
You may combine this 2 path in one for all operation GET, PATH, DELETE,PUT, and remove extra one

**Comment by sravan251** [comment]
fixed lintdiff errors. this may give path removed error in Breaking changes.

❌ R3021 - PathResourceTypeNameCamelCase | Resource type naming must follow camel case. Path: '/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DataLakeAnalytics/accounts/{accountName}/storageAccounts/{storageAccountName}/Containers'New: Microsoft.DataLakeAnalytics/preview/2015-10-01-preview/account.json#L966
-- | --
❌ R3021 - PathResourceTypeNameCamelCase | Resource type naming must follow camel case. Path: '/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DataLakeAnalytics/accounts/{accountName}/storageAccounts/{storageAccountName}/Containers/{containerName}'New: Microsoft.DataLakeAnalytics/preview/2015-10-01-preview/account.json#L1018
❌ R3021 - PathResourceTypeNameCamelCase | Resource type naming must follow camel case. Path: '/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DataLakeAnalytics/accounts/{accountName}/storageAccounts/{storageAccountName}/Containers/{containerName}/listSasTokens'New: Microsoft.DataLakeAnalytics/preview/2015-10-01-preview/account.json#L1074

File: specification/datalake-analytics/resource-manager/Microsoft.DataLakeAnalytics/preview/2015-10-01-preview/account.json [hunk 4]
```diff
@@ -41,509 +37,657 @@
...
+        "x-ms-long-running-operation": true
+      },
+      "put": {
         "tags": [
-          "Account"
+          "Accounts"
         ],
-        "operationId": "Account_GetStorageContainer",
-        "description": "Gets the specified Azure Storage container associated with the given Data Lake Analytics and Azure Storage accounts.",
+        "operationId": "Accounts_Create",
```
[the remark this answers is not in the document]
Comment (sravan251) [reply specification/datalake-analytics/resource-manager/Microsoft.DataLakeAnalytics/preview/2015-10-01-preview/account.json]: updated

File: specification/datalake-analytics/resource-manager/Microsoft.DataLakeAnalytics/preview/2015-10-01-preview/account.json [hunk 5]
```diff
@@ -41,509 +37,657 @@
...
+        "x-ms-long-running-operation": true
+      },
+      "patch": {
         "tags": [
-          "Account"
+          "Accounts"
         ],
-        "operationId": "Account_ListStorageContainers",
-        "description": "Lists the Azure Storage containers, if any, associated with the specified Data Lake Analytics and Azure Storage account combination. The response includes a link to the next page of results, if any.",
+        "operationId": "Accounts_Update",
```
[the remark this answers is not in the document]
Comment (sravan251) [reply specification/datalake-analytics/resource-manager/Microsoft.DataLakeAnalytics/preview/2015-10-01-preview/account.json]: updated

File: specification/datalake-analytics/resource-manager/Microsoft.DataLakeAnalytics/preview/2015-10-01-preview/account.json [hunk 6]
```diff
@@ -596,973 +735,2689 @@
...
-    "/subscriptions/{subscriptionId}/providers/Microsoft.DataLakeAnalytics/accounts": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DataLakeAnalytics/accounts/{accountName}/storageAccounts/{storageAccountName}/containers": {
       "get": {
         "tags": [
-          "Account"
+          "StorageAccounts"
         ],
-        "operationId": "Account_List",
-        "description": "Gets the first page of Data Lake Analytics accounts, if any, within the current subscription. This includes a link to the next page, if any.",
+        "operationId": "StorageAccounts_ListStorageContainers",
```
[the remark this answers is not in the document]
Comment (sravan251) [reply specification/datalake-analytics/resource-manager/Microsoft.DataLakeAnalytics/preview/2015-10-01-preview/account.json]: though i didnt see any operationID changes in Breaking change, I have updated the operationIds based on your comments.

File: specification/datalake-analytics/resource-manager/Microsoft.DataLakeAnalytics/preview/2015-10-01-preview/account.json [hunk 7]
```diff
@@ -596,973 +735,2689 @@
...
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DataLakeAnalytics/accounts/{accountName}/storageAccounts/{storageAccountName}/containers/{containerName}": {
+      "get": {
+        "tags": [
+          "StorageAccounts"
+        ],
+        "operationId": "StorageAccounts_GetStorageContainer",
```
[the remark this answers is not in the document]
Comment (sravan251) [reply specification/datalake-analytics/resource-manager/Microsoft.DataLakeAnalytics/preview/2015-10-01-preview/account.json]: though i didnt see any operationID changes in Breaking change, I have updated the operationIds based on your comments.

File: specification/datalake-analytics/resource-manager/Microsoft.DataLakeAnalytics/preview/2015-10-01-preview/account.json [hunk 8]
```diff
@@ -596,973 +735,2689 @@
...
+            "description": "Successfully deleted the specified Azure Storage account information from the Azure Data Lake Analytics account."
+          },
+          "default": {
             "schema": {
-              "$ref": "#/definitions/DataLakeAnalyticsAccountListResult"
-            }
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
+            },
+            "description": ""
           }
```
[the remark this answers is not in the document]
Comment (sravan251) [reply specification/datalake-analytics/resource-manager/Microsoft.DataLakeAnalytics/preview/2015-10-01-preview/account.json]: updated

File: specification/datalake-analytics/resource-manager/Microsoft.DataLakeAnalytics/preview/2015-10-01-preview/account.json [hunk 9]
```diff
@@ -41,509 +37,657 @@
...
+          },
+          "404": {
+            "description": ""
+          },
+          "default": {
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
+            },
             "description": ""
           }
```
[the remark this answers is not in the document]
Comment (sravan251) [reply specification/datalake-analytics/resource-manager/Microsoft.DataLakeAnalytics/preview/2015-10-01-preview/account.json]: added

**Comment by njuCZ** [comment]
@sravan251, please fix the ModelValidation CI error. About the Swagger Avocado CI, it's false alarm, just ignore it

**Comment by sravan251** [comment]
> @sravan251, please fix the ModelValidation CI error. About the Swagger Avocado CI, it's false alarm, just ignore it

fixed.

**Review by njuCZ — APPROVED** [review]

**Review by JeffreyRichter — APPROVED** [review]
